CYS1.py

```diff
@@ -0,0 +1,753 @@
+            case 'reverse':
+                processed_result.append(reverse(seq))
+            case 'complement':
+                processed_result.append(complement(seq))
+            case 'reverse_complement':
+                processed_result.append(reverse_complement(seq))
+            case 'get_nucl_acid_type':
+                processed_result.append(get_nucl_acid_type(seq))
+            case _:
+                return 'Procedure is not defined in the function'
+
+    if len(processed_result) == 1:
+        return processed_result[0]
+    return processed_result
+
+
+def protein_analysis(*args: str, procedure: str, cell_type: str = None,
+                    letter_format: int = 1) -> list:
+    """
+    Function protein_analysis:
+    - calculates predicted molecular weight of amino acid sequences in kDa
+    (procedure name: molecular_weight)
+    - translate aa sequences from one-letter to three-letter code
+    (procedure name: one_letter_to_three)
+    - calculates total amount of each amino acid in the sequences
+    (procedure name: get_amino_acid_sum)
+    - makes DNA based codon optimization for
+    the introduced amino acid sequences, support 3 types of cells:
+      Esherichia coli, Pichia pastoris, Mouse
+      (procedure name: codon_optimization)
+    - calculates length of amino acid sequences
+    (procedure name: length)
+    - counts the number of atoms of each type in a sequence
+    (procedure name: brutto_count)
+
+    Arguments:
+    - one or multiple string of protein sequences written one letter
+    or three letter code (not mixed)
+    - name of procedure as string
+    - cell type (required only for codon_optimization procedure)
+    - letter_format of code for the protein sequences as int: 1 for one letter,
+     3 for three letter code
+
+    Return:
+    - molecular_weight procedure returns list of floats
+    - one_letter_to_three procedure returns list of strings
+    - get_amino_acid_sum procedure returns list of dictionaries
+    - codon_optimization procedure returns list of strings
+    - length procedure returns list of int values
+    - brutto_count procedure returns list of dictionaries with counts
+    of atoms in the sequence
+    """
+    amino_acid_seqs = name_transform(args, letter_format)
+    procedures = {
+        "molecular_weight": molecular_weight,
+        "one_letter_to_three": one_letter_to_three,
+        "get_amino_acid_sum": get_amino_acid_sum,
+        "codon_optimization": codon_optimization,
+        "length": length,
+        "brutto_count": brutto_count,
+    }
+    if procedure not in procedures.keys():
+        raise ValueError("Requested procedure is not defined")
+    elif procedure == "codon_optimization":
+        return procedures.get(procedure)(amino_acid_seqs, cell_type)
+    else:
+        return procedures.get(procedure)(amino_acid_seqs)
+
+
+def filter_fastq(input_path: str, output_filename: str = '',
+                 gc_bounds: Tuple[int, int] = (0, 100),
```

Comment:
Аргумент gc_bound вполне может быть *float*. Мне кажется, что такие моменты тоже стоит учитывать. Технически, это вообще может быть не только *tuple*, но и просто *int* или *float*, если пользователь указывает только одну границу. 
```suggestion
                 gc_bounds: Union[tuple, int, float] = (0, 100),
```